src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -169,6 +169,23 @@ protected void AssertSelf ()
 			throw new ObjectDisposedException (nameof (AndroidClientHandler));
 		}
 
+		string EncodeUrl (Uri url)
+		{
+			if (url == null)
+				return String.Empty;
+
+			if (String.IsNullOrEmpty (url.Query))
```

Comment:
What about URLs with spaces that lack query strings? For example, `http://www.example.org/hello%20world`? As far as I'm aware, that is a valid URL.